src/main/java/uk/gov/hmcts/reform/judicialapi/service/impl/JudicialUserServiceImpl.java

```diff
@@ -396,7 +397,9 @@ private AuthorisationRefreshResponse buildAuthorisationRefreshResponseDto(
     }
 
     private List<String> getRoleIdList(List<JudicialRoleType> judicialRoleTypes) {
-        return judicialRoleTypes.stream().map(JudicialRoleType::getTitle).collect(Collectors.toList());
+        return judicialRoleTypes.stream()
+                .filter(e -> e.getEndDate() == null || !e.getEndDate().isBefore(LocalDateTime.now()))
```

Comment:
Can this filter be done at the  database query level? @amitkumarshrivastava @sahitya-desireddy

Reply:
@SujeethKumarAri  no because in the query we r just retrieving the valid user profiles but only in the code we are using getter to retrieve all authorisations and appointments for that user profile so will work if added in the code to get active roles ..Thank you ...

Reply:
thanks @sahitya-desireddy for your response.
Can't we add the filter in here - https://github.com/hmcts/rd-judicial-api/blob/RDCC-3643/src/main/java/uk/gov/hmcts/reform/judicialapi/repository/UserProfileRepository.java#L93 and all other queries? you can try that locally in the database and see if you can filter it out

Reply:
Hi @SujeethKumarAri  
@Query(value = "select distinct per "
            + "from judicial_user_profile per "
          .....
            + "where (per.objectId != '' and per.objectId is not null) "
            + "and ((appt.endDate >= CURRENT_DATE or appt.endDate is null) "
            + "or (auth.endDate >= CURRENT_DATE or auth.endDate is null)) ")

here query returning  judicial_user_profile. if we add role filter in where condition it will filter profile not the roles.
for ex: we are filtering with appointment and authorisation, even though we are getting expired appointments and authorisations in the response.

I tried locally with this repository to fix, but it didn't work.

Reply:
ah ok, my bad I misunderstood. So we are filtering the roles to show active roles not profiles based on the active roles. correct @sahitya-desireddy @amitkumarshrivastava @kavithagade12 ?

Reply:
yes @SujeethKumarAri